### PR TITLE
Add go frameworks to docs

### PR DESCRIPTION
**Pull request by owen-mc**

### CODE REVIEW

File: docs/language/support/reusables/frameworks.rst [hunk 1]
```diff
@@ -18,6 +18,46 @@ C# built-in support
...
+   Name, Category
+   SendGrid, Email library
+   json-iterator, Serialization
+   Gin, Web framework
+   glog, Logging library
+   klog, Logging library
+   go-restful, Web application framework
+   Logrus, Logging library
+   Macaron, Web framework
+   mux, http request router and dispatcher
```
Comment (max-schaefer) [inline docs/language/support/reusables/frameworks.rst:38]:
```suggestion
   mux, HTTP request router and dispatcher
```

**Review by max-schaefer — CHANGES_REQUESTED** [review]
LGTM modulo one small suggestion. Would it make sense to sort alphabetically to make it easier to find a specific library?

File: docs/language/support/reusables/frameworks.rst [hunk 2]
```diff
@@ -18,6 +18,46 @@ C# built-in support
...
+   sqlx, Database
+   go-sh, Utility library
+   golang.org/x/crypto/ssh, Network communicator
+   golang.org/x/net/websocket, Network communicator
+   Gorilla websocket, Network communicator
+   ws, Network communicator
+   nhooyr.io/websocket, Network communicator
+   GoWebsocket, Network communicator
+   ws, Network communicator
+   https://github.com/antchfx/xpath, Xpath library
```
Comment (intrigus-lgtm) [inline docs/language/support/reusables/frameworks.rst:52]:
```suggestion
   https://github.com/antchfx/xpath, XPath library
```
Isn't it usually capitalized like this: `XPath`?
See the other instances below.

File: docs/language/support/reusables/frameworks.rst [hunk 3]
```diff
@@ -18,6 +18,46 @@ C# built-in support
...
+   Macaron, Web framework
+   mongo, Database
+   mux, HTTP request router and dispatcher
+   nhooyr.io/websocket, Network communicator
+   pg, Database
+   sqlx, Database
+   SendGrid, Email library
+   Squirrel, Database
+   ws, Network communicator
+   ws, Network communicator
```
Comment (max-schaefer) [inline docs/language/support/reusables/frameworks.rst:58]: Duplicate entry.

Comment (owen-mc) [reply docs/language/support/reusables/frameworks.rst]: For a minute I thought I'd messed up my sorting, but it turns out the duplicate entry was already there and sorting just made it obvious

**Review by max-schaefer — APPROVED** [review]

**Pull request merged**